openvino_tensorflow/ovtf_builder.cc

```diff
@@ -711,21 +713,24 @@ static Status TranslateAvgPoolOp(const Node* op,
   TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "padding", &tf_padding_type));
   TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "data_format", &tf_data_format));
 
-  if (tf_data_format != "NHWC" && tf_data_format != "NCHW") {
+  if ((tf_data_format != "NHWC") && (tf_data_format != "NCHW") &&
+      (tf_data_format != "NDHWC")) {
     return errors::InvalidArgument(
-        "AvgPool data format is neither NHWC nor NCHW");
+        "AvgPool data format is none of NHWC, NCHW, or NDHWC");
   }
 
-  bool is_nhwc = (tf_data_format == "NHWC");
+  bool is_nhwc = (tf_data_format == "NHWC") || (tf_data_format == "NDHWC");
 
   OVTF_VLOG(3) << ng::join(tf_strides);
   OVTF_VLOG(3) << ng::join(tf_ksize);
   OVTF_VLOG(3) << tf_padding_type;
   OVTF_VLOG(3) << tf_data_format;
 
-  ng::Strides ng_strides(2);
-  ng::Shape ng_image_shape(2);
-  ng::Shape ng_kernel_shape(2);
+  ng::Strides ng_strides(N);
+  ng::Shape ng_image_shape(N);
+  ng::Shape ng_kernel_shape(N);
+  ng::Shape ng_dilations{N, 1};
```

Comment:
Is N required?

Reply:
Removed this part of the code which belonged to the older method of padding calculation. Now padding is taken care by OpenVINO's auto_pad feature.